lib/http.js

```diff
@@ -31,11 +31,22 @@ exports.request = function(options, cb) {
   return new ClientRequest(options, cb);
 };
 
-exports.get = function(options, cb) {
-  var req = exports.request(options, cb);
-  req.end();
-  return req;
-};
+function shortcutMethod(method) {
+  return function(options, cb) {
+    options['method'] = method;
+    var req = exports.request(options, cb);
```

Comment:
So something like this, except one which actually populates the callback response properly:

```
function httpF(method) {
    return function (options, body_or_cb, cb) {
        if (!cb) {
            cb = body_or_cb;
            body_or_cb = null;
        }
        options['method'] = method;
        if (body_or_cb)
            if (!options)
                options = { 'headers': { 'Content-Length': Buffer.byteLength(body_or_cb) } };
            else if (!options.headers)
                options.headers = { 'Content-Length': Buffer.byteLength(body_or_cb) };
            else if (!options.headers['Content-Length'])
                options.headers['Content-Length'] = Buffer.byteLength(body_or_cb);
        var req = http.request(options, cb);
        //body_or_cb ? req.end(<string>body_or_cb, cb) : req.end();
        body_or_cb && req.write(body_or_cb);
        req.end();
        return req;
    };
}
```